rhino/src/main/java/org/mozilla/javascript/MemberBox.java

```diff
@@ -111,6 +109,23 @@ List<TypeInfo> getArgTypes() {
         return argTypeInfos;
     }
 
+    public NullabilityDetector.NullabilityAccessor getArgNullability() {
+        var got = this.argNullability;
+        if (got == null) {
+            synchronized (this) {
```

Comment:
What's the idea of making this synchronized? (using double-checked locking, which is OK since argNullability is volatile)? It looks to me like we need it in the event that multiple threads are accessing the same class, but I want to make sure -- there are other places in Rhino where we sprinkle synchronized blocks around and it's not always clear to me that they actually are necessary to enable thread-safety.